contrib/inventory/cloudforms.py

```diff
@@ -0,0 +1,351 @@
+                            help='Force refresh of cache by making API requests to CloudForms (default: False - use cache files)')
+        parser.add_argument('--debug', action='store_true', default=False, help='Show debug output while running (default: False)')
+        self.args = parser.parse_args()
+
+    def _get_json(self, url):
+        """
+        Make a request and return the JSON
+        """
+        results = []
+
+        ret = requests.get(url,
+                           auth=HTTPBasicAuth(self.cloudforms_user, self.cloudforms_pw),
+                           verify=self.cloudforms_ssl_verify)
+
+        ret.raise_for_status()
+        results = json.loads(ret.text)
+
+        if self.args.debug:
+            print "======================================================================="
+            print "======================================================================="
+            print "======================================================================="
+            print json.dumps(results, indent=2)
+            print "======================================================================="
+            print "======================================================================="
+            print "======================================================================="
+
+        return results
+
+    def _get_hosts(self):
+        """
+        Get all hosts by paging through the results
+        """
+        limit = self.cloudforms_limit
+
+        page = 0
+        last_page = False
+
+        results = []
+
+        while not last_page:
+            offset = page * limit
+            ret = self._get_json("%s/api/vms?offset=%s&limit=%s&expand=resources,tags&attributes=id,ipaddresses,name,power_state" % (self.cloudforms_url, offset, limit))
+            results += ret['resources']
+            if ret['subcount'] < limit:
+                last_page = True
+            page += 1
+
+        return results
+
+    def _get_host_by_id(self, hid):
+        return self._get_json("%s/api/vms/%s" % (self.cloudforms_url, hid))
+
+    def _resolve_host(self, host):
+        """
+        Resolve host details.
+        """
+        host_id = host['id']
+        ret = self._get_host_by_id(host_id)
+
+        if self.cloudforms_purge_actions:
+            del ret['actions']
+
+        return ret
+
+    def update_cache(self):
+        """
+        Make calls to cloudforms and save the output in a cache
+        """
+        self.groups = dict()
+        self.hosts = dict()
+
+        if self.args.debug:
+            print "Updating cache..."
+
+        for host in self._get_hosts():
+            if host['power_state'] != 'on':
+                if self.args.debug:
+                    print "Skipping %s because power_state = %s" % (host['name'], host['power_state'])
+                continue
+
+            if self.cloudforms_prefer_ip_address and host.has_key('ipaddresses') and host['ipaddresses'][0]:
```

Comment:
Instead of changing the `name`, can we set `ansible_ssh_host` to the discovered `ipaddress`?